### DISCUSSION TITLE
SecurityHub_ListFinding_Response

### DISCUSSION THREAD

**Discussion by niraj-ranjan-comprinno** [aws/aws-sdk-java-v2#3888] [Q&A]
### Describe the bug

SecurityHubClient awsSecurityHubAsyncClientBuilder = SecurityHubClient.builder()
				.credentialsProvider(new AWSStaticCredentialsProvider(credentials)).region(Helper.getRegion(regionName))
				.build();

awsSecurityHubAsyncClientBuilder.getFindings(getFindingsRequest.build())

Above api gives list of security hub findings.

In any element of array, 
 there is a field called productFields -> StandardsControlArn

And now this field is not coming.

### Expected Behavior

In list findings response, 
Field called productFields -> StandardsControlArn  value must present.

So that, We can find how many findings are passed/failed for a particular standard control.

### Current Behavior

In list findings response, 
Field called productFields -> StandardsControlArn  value is not present.

### Reproduction Steps

SecurityHubClient awsSecurityHubAsyncClientBuilder = SecurityHubClient.builder()
				.credentialsProvider(new AWSStaticCredentialsProvider(credentials)).region(Region.AP_EAST_1)
				.build();

List<AwsSecurityFinding> awsSecurityFinding = awsSecurityHubAsyncClientBuilder.getFindings(GetFindingsRequest.builder().build()).findings()

This will give you list of findings.

You can check any one of item from the above reponse.

### AWS Java SDK version used

2.17.271

### JDK version used

OpenJDK Runtime Environment Corretto-17.0.2.8.1 (build 17.0.2+8-LTS)

### Operating System and version

Mac

**Comment by debora-ito** [comment]
> Field called productFields -> StandardsControlArn value must present.

According to the [API Reference](https://docs.aws.amazon.com/securityhub/1.0/APIReference/API_AwsSecurityFinding.html),  "productFields" can contain up to 50 key-value pairs. So the presence of StandardsControlArn is not guaranteed for every finding on the list.

**Comment by niraj-ranjan-comprinno** [comment]
Ok, If that case then how to find productFields => StandardsControlArn for security hub finding.

**Comment by debora-ito** [comment]
I don't have expertise in Security Hub, so, again, I need to refer to their documentation.

According to the Security Hub [User Guide](https://docs.aws.amazon.com/securityhub/latest/userguide/controls-findings-create-update.html#control-findings-asff-productfields), StandardsControlArn is removed if you turn on consolidated control findings.

**Comment by debora-ito** [comment]
I'll move this to the Github Discussion tab, since these are usage questions.
